DM-26646: Make endpoint manager monitor future day(s)

mxk62 opened the pull request:
Files from a single observation night can be placed in different directories and these directories can be created in a time zone
different from the one the endpoint site operates in.  Though the endpoint manager was already able to monitor past days, it is not able to look "into the future" which may be required for some time zones.  To address this issue I added an option to the search function parsing rsync logs which will allow one to instruct the manager to "look ahead" of local time for a given number of days (1, by default).

mxk62 committed:
Make endpoint manager monitor future day(s)

Files from a single observation night can be placed in different
directories and these directories can be created in a time zone
different from the one the endpoint site operates in.  Though the
endpoint manager was already able to monitor past days, it is not able
to look "into the future" which may be required for some time zones.  To
address this issue I added an option to the search function parsing
rsync logs which will allow one to instruct the manager to "look ahead"
of local time for a given number of days (1, by default).

MichelleGower commented on python/lsst/dbb/buffmngrs/endpoint/finder/finder.py:
```diff
@@ -273,6 +273,12 @@ def parse_rsync_logs(directory, blacklist=None,
+    As the files from a single observation night can be placed in different
+    directories and these directories can be created in a time zone different
+    from the one the endpoint site operates in, be default the function
+    monitors also log files in directories corresponding to a day before and
```
Why is it "a day before and a day after"?  Don't the past_days and future_days values determine how many days before and after?

mxk62 replied:
Yes, at the end of a day (no pun intended) the actual time span is determined by `past_days` and `future_days` settings, but I wrote

> **by default** the function [...]

I also see that I misspelled "by" (will fix), but I think you get my point.

Anyway, I'll add a sentence clarifying that this interval can be adjusted with `past_days` and `future_days`.

MichelleGower commented on python/lsst/dbb/buffmngrs/endpoint/finder/finder.py:
```diff
@@ -304,11 +311,12 @@ def parse_rsync_logs(directory, blacklist=None,
+        origin = isodate
+    past = [origin - timedelta(days=n) for n in range(1, past_days+1)]
+    future = [origin + timedelta(days=n) for n in range(1, future_days+1)]
+    for day in sorted([*past, origin, *future]):
```
Didn't see anywhere where past and future are used separately.  Can't the day list be made in single loop and without sorting?  Or even just loop on day offset making the particular date inside the loop since the list of dates doesn't seem to be used elsewhere.

MichelleGower reviewed (approved):
Minor changes and then can merge.

mxk62 committed:
Simplify generating date range

mxk62 committed:
Update validation schema

Validation schema for Finder's configuration files was out of date.  I
changed it to reflect the recent changes.

mxk62 committed:
Improve parse_rsync_logs() docstring
